### PR TITLE
Feature/s21_strncmp

### LINKED ISSUE #24: `s21_strncmp` - Compares up to `n` bytes of `str1` and `str2`.

### PR DESCRIPTION
**Pull request by DarkDomian**
## Closed: #24 | Implement `s21_strncmp()` function

**Changes**  
✨ Added `s21_strncmp()` implementation:
- Compares up to n characters of two strings
- Returns 0 if equal, positive if str1 > str2, negative if str1 < str2
- Handles all edge cases identically to standard `strncmp()`

🧪 Added comprehensive test coverage:
- 9 test cases covering all edge conditions
- Byte-by-byte comparison verification
- ASCII compatibility checks

📚 Updated documentation:
- Added detailed function description in `s21_string.h`
- Specified return value behavior

## Verification  

- [ ] All tests pass (Check library)  
- [ ] 100% line coverage (gcov)  
- [ ] Valgrind-clean (no memory leaks)  
- [ ] Clang-format compliant  

## Example Usage
```c
int result = s21_strncmp("hello", "hellp", 5);
// Returns negative value
```

## Testing:
```bash
make style-check
make strncmp.test
```

### CODE REVIEW

File: src/s21_strncmp.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include "../headers/s21_string.h"
+
+int s21_strncmp(const char *str1, const char *str2, s21_size_t n) {
+  unsigned char u1, u2;
```
Comment (ParthenGithub) [inline src/s21_strncmp.c]:
```c
src/s21_strncmp.c:4:17: style: The scope of the variable 'u1' can be reduced. [variableScope]
  unsigned char u1, u2;
                ^
src/s21_strncmp.c:4:21: style: The scope of the variable 'u2' can be reduced. [variableScope]
  unsigned char u1, u2;
                    ^
```

IDK how to fix, figure out yourself

**Review by ParthenGithub — CHANGES_REQUESTED** [review]

**Review by ParthenGithub — APPROVED** [review]

**Review by gashturik — APPROVED** [review]
